File: Reveal Cards In Increasing Order.py
class Solution:
    def deckRevealedIncreasing(self, deck: List[int]) -> List[int]:
        n = len(deck)
        deck.sort()
        i = n-2
        q = deque([deck[-1]])
        while i >= 0:
            q.appendleft(q.pop())
            q.appendleft(deck[i])
            i -= 1
        return q

File: Reveal Cards In Increasing Order.cpp
class Solution {
public:
    vector<int> deckRevealedIncreasing(vector<int>& deck) {
        int n = deck.size();
        sort(deck.begin(), deck.end());
        int i = n-2;
        deque<int> q;
        q.push_back(deck[n-1]);
        while (i >= 0) {
            q.push_front(q.back());
            q.push_front(deck[i]);
            q.pop_back();
            i--;
        }
        vector<int> ans;
        ans.reserve(n);
        for (int& num: q) {
            ans.push_back(num);
        }
        return ans;
    }
};
